services/contract_update.proto

```diff
@@ -32,75 +41,93 @@ import "timestamp.proto";
 
 /**
- * At consensus, updates the fields of a smart contract to the given values.
- * 
- * If no value is given for a field, that field is left unchanged on the contract. For an immutable
- * smart contract (that is, a contract created without an adminKey), only the expirationTime may be
- * updated; setting any other field in this case will cause the transaction status to resolve to
- * MODIFYING_IMMUTABLE_CONTRACT.
- * 
- * --- Signing Requirements ---
- * 1. Whether or not a contract has an admin key, its expiry can be extended with only the
- *    transaction payer's signature.
- * 2. Updating any other field of a mutable contract requires the admin key's signature.
- * 3. If the update transaction includes a new admin key, this new key must also sign <b>unless</b>
- *    it is exactly an empty <tt>KeyList</tt>. This special sentinel key removes the existing admin
- *    key and causes the contract to become immutable. (Other <tt>Key</tt> structures without a
- *    constituent <tt>Ed25519</tt> key will be rejected with <tt>INVALID_ADMIN_KEY</tt>.)
- * 4. If the update transaction sets the AccountID auto_renew_account_id wrapper field to anything
- *    other than the sentinel <tt>0.0.0</tt> value, then the key of the referenced account must sign.
+ * Modify the current state of a smart contract.
+ *
+ * ### Requirements
+ * - The `adminKey` MUST sign all contract update transactions except one
+ *   that only updates the `expirationTime`.
+ * - A transaction that modifies any field other than `expirationTime` for
+ *   a contract without a valid `adminKey` set SHALL fail with response
+ *   code `MODIFYING_IMMUTABLE_CONTRACT`.
+ * - Fields set to non-default values in this transaction SHALL be updated on
+ *   success. Fields not set to non-default values SHALL NOT be
+ *   updated on success.
+ *
+ * ### Block Stream Effects
+ * None
  */
 message ContractUpdateTransactionBody {
     /**
-     * The id of the contract to be updated
+     * The contact ID that identifies the smart contract to be updated.<br/>
+     * This field MUST be set, and MUST NOT be a default ID (`0.0.0`).
      */
     ContractID contractID = 1;
 
     /**
-     * The new expiry of the contract, no earlier than the current expiry (resolves to
-     * EXPIRATION_REDUCTION_NOT_ALLOWED otherwise)
+     * If set, modify the time at which this contract will expire.<br/>
+     * An expired contract requires a rent payment to "renew" the contract.
+     * A transaction to update this field is how that rent payment is made.
+     * <p>
+     * This value MUST NOT be less than the current `expirationTime`
+     * of the contract. If this value is earlier than the current
+     * value, the transaction SHALL fail with response
+     * code `EXPIRATION_REDUCTION_NOT_ALLOWED`.
      */
     Timestamp expirationTime = 2;
 
     /**
-     * The new key to control updates to the contract
+     * If set, modify the key that authorizes updates to the contract.
+     * <p>
```

Comment:
Here we have a <p> tag but in the previous field we have a <br/> ?

Reply:
This separates summary from requirements.  The prior field requires more description than the summary statement, so it includes a descriptive sentence; this field does not require that additional descriptive sentence.